src/Symfony/Component/Cache/Traits/PhpArrayTrait.php

```diff
@@ -65,36 +68,25 @@ public function warmUp(array $values)
 
 EOF;
 
+        $exportSerializer = new PhpExportSerializer();
+
         foreach ($values as $key => $value) {
             CacheItem::validateKey(\is_int($key) ? (string) $key : $key);
 
-            if (null === $value || \is_object($value)) {
-                try {
-                    $value = serialize($value);
-                } catch (\Exception $e) {
-                    throw new InvalidArgumentException(sprintf('Cache key "%s" has non-serializable %s value.', $key, get_class($value)), 0, $e);
-                }
-            } elseif (\is_array($value)) {
+            try {
                 try {
-                    $serialized = serialize($value);
-                    $unserialized = unserialize($serialized);
-                } catch (\Exception $e) {
-                    throw new InvalidArgumentException(sprintf('Cache key "%s" has non-serializable array value.', $key), 0, $e);
+                    $exportedValue = $exportSerializer->serialize($value);
+                    $valuePart = 'array("isSerialized" => false, "value" => '.$exportedValue.')';
```

Comment:
For me, an important criteria of this PR should be to preserve the original data format, so that we keep compatibility with existing dumped values. Here, this change it, and adds overhead to the storage, filling it with medadata that wasn't needed before.

I think the opcache-base storages should just not allow configuration of the serializer. WDYT?

Reply:
Previous realization favors scalar and arrays of scalar data. Nulls and objects was stored in serialized form. It adds performance hit on every cache read or forces to use opcache for primitive types only.
It is possible to store values directly, but all checks in form: ```isset($this->values, $key)``` should be rewritten.

Reply:
> Previous realization favors scalar and arrays of scalar data. Nulls and objects was stored in serialized form

that's intended: only these benefit from opcache's shared memory

> It adds performance hit on every cache read or forces to use opcache for primitive types only

do you have numbers about that to illustrate this statement?

Reply:
Test: https://gist.github.com/palex-fpt/82fc3deed09c2de2a9023080a9613ce3

master results:
Duration (first/next) (largeArrayOfNulls): 14.200/2.300 s
Duration (first/next) (largeArrayOfStrings): 12.200/2.500 s
Duration (first/next) (largeArrayOfSmallObjects): 18.600/5.000 s
Duration (first/next) (largeObject): 130.400/112.300 s

branch results:
Duration (first/next) (largeArrayOfNulls): 16.500/3.500 s
Duration (first/next) (largeArrayOfStrings): 11.300/2.800 s
Duration (first/next) (largeArrayOfSmallObjects): 30.200/5.200 s
Duration (first/next) (largeObject): 145.700/3.400 s
Duration (first/next) (largeExportableObject): 7.500/3.300 s